File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s=set(nums)
        r=0
        for i in s:
            if (i-1) not in s:
                l=1
                j=i
                while True:
                    if j+1 in s:
                        l=l+1
                        j=j+1
                    else :
                        break
                if r<l:
                    r=l
            else:
                continue 
        return r
        

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> prefix_prod;
        vector<int> suffix_prod(nums.size());
        prefix_prod.push_back(1);
        for (int i =1;i<nums.size();i++){
            int prev = prefix_prod[i-1];
            prefix_prod.push_back(prev * nums[i-1]);
        }

        suffix_prod[nums.size() - 1] = 1;
        for (int i =nums.size() -2;i>=0;i--){
            int suff = suffix_prod[i+1];
            suffix_prod[i] = suff * nums[i+1];
        }

        for(int i=0;i<nums.size();i++){
            nums[i] = prefix_prod[i] * suffix_prod[i];
        }

        return nums;

    }
};
